File: src/main/java/org/jdc/template/ux/individual/IndividualContract.java
package org.jdc.template.ux.individual;

import org.jdc.template.model.database.main.individual.Individual;

public class IndividualContract {
    interface View {
        void showIndividual(Individual individual);

        void close();

        void showEditIndividual(long individualId);

        void promptDeleteIndividual();
    }

    public class Extras {
        public static final String EXTRA_ID = "INDIVIDUAL_ID";
    }
}

File: src/main/kotlin/org/jdc/template/ux/individual/IndividualContract.kt
package org.jdc.template.ux.individual

import android.content.Intent
import me.eugeniomarletti.extras.ActivityCompanion
import me.eugeniomarletti.extras.intent.IntentExtra
import me.eugeniomarletti.extras.intent.base.Long
import org.jdc.template.model.database.main.individual.Individual

class IndividualContract {
    interface View {
        fun showIndividual(individual: Individual)
        fun close()
        fun showEditIndividual(individualId: Long)
        fun promptDeleteIndividual()
    }

    companion object : ActivityCompanion<IntentOptions>(IntentOptions, IndividualActivity::class)

    object IntentOptions {
        var Intent.individualId by IntentExtra.Long(defaultValue = 0L)
    }
}
